pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -185,7 +190,7 @@ func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcil
 	runningPhase := "Running"
 	if machine.Status.Phase == nil {
 		// Phase is nil and should be ignored by WMCO until phase is set
-		return reconcile.Result{}, nil
+		return reconcile.Result{}, fmt.Errorf("could not get the phase associated with machine %s", machine.Name)
```

Comment:
Why should we requeue? Isnt this going to be caught when the machine's status is updated? We will get another event when that happens

Reply:
I dont understand the value of this. How much long does the event take to happen? This change means that we will have an extra reconciliation as we will have to handle the phase change event anyway.

Reply:
Nope that will be discarded..

Reply:
It will be discarded only after we pull the machine object and do all the checks to make sure that nothing has to be done.

But still that doesnt answer the question of 
> How much long does the event take to happen?

I dont see why we should constantly try and requeue for something that can only be done once the machine object changes state. Being notified of when the machine object changes state is the purpose of a controller.

Reply:
> It will be discarded only after we pull the machine object and do all the checks to make sure that nothing has to be done.

Nope, work queues work based on sets. If the dirty set is already having the [event](https://openkruise.io/en-us/blog/blog2.html), the new event will be discarded. 

>  How much longer does the event take to happen?

It takes the amount of time we wait for wait for another reconcile which is 5 seconds I guess or if the update event happens immediately after reconcile. It's not about the time it takes for how long to happen the question is why should WMCO call an object is successfully reconciled and wait for another event to occur when it knows that this event should have been requeued.

Reply:
> If the dirty set is already having the event, the new event will be discarded
ok that makes sense.

> why should WMCO call an object is successfully reconciled
Because we have successfully reconciled based on how the machine object is. The machine object does not yet have the correct fields for us to do anything further. We have done what we can with the object and can wait until the object is updated once more

Reply:
> Because we have successfully reconciled based on how the machine object is. The machine object does not yet have the correct fields for us to do anything further. We have done what we can with the object and can wait until the object is updated once more

I see that as WMCO reacting too early. Ideally WMCO shouldn't have entered reconciler if the machine phase is not nil. However if you feel strongly about it, I can remove this check and we can come back and discuss about it offline.

Reply:
discussed this, i am fine with this currently as in the future we will not reach reconcile with this information not set